sweepbatcher/sweep_batcher.go

```diff
@@ -661,87 +699,123 @@ func (b *Batcher) writeToErrChan(ctx context.Context, err error) error {
 func (b *Batcher) convertSweep(ctx context.Context, dbSweep *dbSweep) (
 	*sweep, error) {
 
-	swap, err := b.swapStore.FetchLoopOutSwap(ctx, dbSweep.SwapHash)
+	s, err := b.sweepStore.FetchSweep(ctx, dbSweep.SwapHash)
 	if err != nil {
-		return nil, fmt.Errorf("failed to fetch loop out for %x: %w",
+		return nil, fmt.Errorf("failed to fetch sweep data for %x: %v",
 			dbSweep.SwapHash[:6], err)
 	}
 
-	htlc, err := utils.GetHtlc(
-		dbSweep.SwapHash, &swap.Contract.SwapContract, b.chainParams,
-	)
-	if err != nil {
-		return nil, err
-	}
-
-	swapPaymentAddr, err := utils.ObtainSwapPaymentAddr(
-		swap.Contract.SwapInvoice, b.chainParams,
-	)
-	if err != nil {
-		return nil, err
-	}
-
 	return &sweep{
-		swapHash:               swap.Hash,
+		swapHash:               dbSweep.SwapHash,
 		outpoint:               dbSweep.Outpoint,
 		value:                  dbSweep.Amount,
-		confTarget:             swap.Contract.SweepConfTarget,
-		timeout:                swap.Contract.CltvExpiry,
-		initiationHeight:       swap.Contract.InitiationHeight,
-		htlc:                   *htlc,
-		preimage:               swap.Contract.Preimage,
-		swapInvoicePaymentAddr: *swapPaymentAddr,
-		htlcKeys:               swap.Contract.HtlcKeys,
-		htlcSuccessEstimator:   htlc.AddSuccessToEstimator,
-		protocolVersion:        swap.Contract.ProtocolVersion,
-		isExternalAddr:         swap.Contract.IsExternalAddr,
-		destAddr:               swap.Contract.DestAddr,
+		confTarget:             s.ConfTarget,
+		timeout:                s.Timeout,
+		initiationHeight:       s.InitiationHeight,
+		htlc:                   s.HTLC,
+		preimage:               s.Preimage,
+		swapInvoicePaymentAddr: s.SwapInvoicePaymentAddr,
+		htlcKeys:               s.HTLCKeys,
+		htlcSuccessEstimator:   s.HTLCSuccessEstimator,
+		protocolVersion:        s.ProtocolVersion,
+		isExternalAddr:         s.IsExternalAddr,
+		destAddr:               s.DestAddr,
 	}, nil
 }
 
-// fetchSweep fetches the sweep related information from the database.
-func (b *Batcher) fetchSweep(ctx context.Context,
-	sweepReq SweepRequest) (*sweep, error) {
+// LoopOutFetcher is used to load LoopOut swaps from the database.
+// It is implemented by loopdb.SwapStore.
+type LoopOutFetcher interface {
+	// FetchLoopOutSwap returns the loop out swap with the given hash.
+	FetchLoopOutSwap(ctx context.Context,
+		hash lntypes.Hash) (*loopdb.LoopOut, error)
+}
 
-	swapHash, err := lntypes.MakeHash(sweepReq.SwapHash[:])
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse swapHash: %v", err)
-	}
+// SwapStoreWrapper is LoopOutFetcher wrapper providing SweepFetcher interface.
+type SwapStoreWrapper struct {
+	// swapStore is used to load LoopOut swaps from the database.
+	swapStore LoopOutFetcher
 
-	swap, err := b.swapStore.FetchLoopOutSwap(ctx, swapHash)
+	// chainParams are the chain parameters of the chain that is used by
+	// batches.
+	chainParams *chaincfg.Params
```

Comment:
`chaincfg` is used by method `SwapStoreWrapper.FetchSweep` in calls to `utils.GetHtlc` and `utils.ObtainSwapPaymentAddr`. We can't reuse Batcher's `chaincfg` there, because SwapStoreWrapper does not have a reference to Batcher object.